contents/tutorials/ios-session-replays.md

```diff
@@ -0,0 +1,190 @@
+---
+title: How to set up iOS session replays
+date: 2024-07-10
+author:
+  - lior-neu-ner
+tags:
+  - session replay
+---
+
+[Session replays](/session-replay) are a useful support tool for understanding how users are interacting with your iOS app. It also helps you debug and recreate issues. 
+
+To show how you to set it up with PostHog, in this tutorial we create a basic SwiftUI app, add PostHog, and [enable session recordings](/docs/session-replay/mobile#ios).
+
+## 1. Create a basic iOS app
+
+Our sample app will have two screens:
+
+- The first screen is be a `login` screen with email and password textfields.
+- The second screen is a simple screen with welcome text and logout button.
+
+The first step is to create a new app. Open XCode and click **Create new project**. Select iOS as your platform, then **App** and press **next**. Give your app a name, select `SwiftUI` as the interface, and the defaults for everything else. Click next and then **Create**.
+
+Then, replace your code in `ContentView.swift` with the following:
+
+```swift file=ContentView.swift
+import SwiftUI
+
+struct ContentView: View {
+    @State private var email: String = ""
+    @State private var password: String = ""
+    @State private var isLoggedIn: Bool = false
+
+    var body: some View {
+        NavigationView {
+            if isLoggedIn {
+                WelcomeView(isLoggedIn: $isLoggedIn)
+            } else {
+                VStack {
+                    TextField("Email", text: $email)
+                        .padding()
+                        .keyboardType(.emailAddress)
+                        .autocapitalization(.none)
+                        .border(Color.gray)
+                    
+                    SecureField("Password", text: $password)
+                        .padding()
+                        .border(Color.gray)
+                    
+                    Button(action: {
+                        // Simple login logic
+                        if !email.isEmpty && !password.isEmpty {
+                            isLoggedIn = true
+                        }
+                    }) {
+                        Text("Login")
+                            .frame(maxWidth: .infinity)
+                            .padding()
+                            .background(Color.blue)
+                            .foregroundColor(.white)
+                            .cornerRadius(10)
+                    }
+                    .padding(.top, 20)
+                }
+                .padding()
+                .navigationTitle("Login")
+            }
+        }
+    }
+}
+
+struct WelcomeView: View {
+    @Binding var isLoggedIn: Bool
+    
+    var body: some View {
+        VStack {
+            Text("Welcome!")
+                .accessibilityIdentifier("ph-no-capture")
```

Comment:
This one and L170 won't work, see the comment here https://posthog.com/docs/session-replay/mobile#masking-and-redaction-1

Reply:
Will wait for this issue to be fixed before shipping

Reply:
@marandaneto Im back from PTO. Any update on this issue?

Reply:
sadly no, I gave another stab at it yesterday https://github.com/PostHog/posthog-ios/issues/162#issuecomment-2310133365 and could not figure out any other way so far.
I am currently debugging some other issues with a bit more high prior but I will come back to it.
My recommendation for now is to either just mention this is not supported for swiftUI right now but it is for UIKit, or convert the example to use UIKit which is still very popular.
I believe one of the 2 options is better than not having a guide at all.

Reply:
Okay thanks. I will give it a few weeks before making a call on which direction to go